deployer/src/proxy.rs

```diff
@@ -97,7 +87,19 @@ pub async fn handle(
         }
     };
 
-    match reverse_proxy(remote_address.ip(), &proxy_address.to_string(), req).await {
+    let client = match req.version() {
+        Version::HTTP_10 | Version::HTTP_11 => &H1_PROXY_CLIENT,
+        Version::HTTP_2 => &H2_PROXY_CLIENT,
+        protocol => {
+            error!(protocol = ?protocol, "received request with unsupported protocol");
+            return Ok(Response::builder()
+                .status(StatusCode::INTERNAL_SERVER_ERROR)
```

Comment:
Should this not be Bad Request?

Reply:
Oh, thanks! Actually I think it should be [505](https://httpwg.org/specs/rfc9110.html#status.505 with a message detailing which versions are supported.